// AUTOMATICALLY GENERATED BY RCPPDEEPSTATE PLEASE DO NOT EDIT BY HAND, INSTEAD EDIT
// BayesA_DeepState_TestHarness_generation.cpp and BayesA_DeepState_TestHarness_checks.cpp

#include <fstream>
#include <RInside.h>
#include <iostream>
#include <RcppDeepState.h>
#include <qs.h>
#include <DeepState.hpp>

SEXP BayesA(NumericVector y, NumericMatrix X, double it, double bi, double df, double R2);

TEST(bWGR_deepstate_test,BayesA_test){
  RInside R;
  std::cout << "input starts" << std::endl;
  NumericVector y  = RcppDeepState_NumericVector();
  qs::c_qsave(y,"/home/akhila/fuzzer_packages/fuzzedpackages/bWGR/inst/testfiles/BayesA/inputs/y.qs",
		"high", "zstd", 1, 15, true, 1);
  std::cout << "y values: "<< y << std::endl;
  NumericMatrix X  = RcppDeepState_NumericMatrix();
  qs::c_qsave(X,"/home/akhila/fuzzer_packages/fuzzedpackages/bWGR/inst/testfiles/BayesA/inputs/X.qs",
		"high", "zstd", 1, 15, true, 1);
  std::cout << "X values: "<< X << std::endl;
  NumericVector it(1);
  it[0]  = RcppDeepState_double();
  qs::c_qsave(it,"/home/akhila/fuzzer_packages/fuzzedpackages/bWGR/inst/testfiles/BayesA/inputs/it.qs",
		"high", "zstd", 1, 15, true, 1);
  std::cout << "it values: "<< it << std::endl;
  NumericVector bi(1);
  bi[0]  = RcppDeepState_double();
  qs::c_qsave(bi,"/home/akhila/fuzzer_packages/fuzzedpackages/bWGR/inst/testfiles/BayesA/inputs/bi.qs",
		"high", "zstd", 1, 15, true, 1);
  std::cout << "bi values: "<< bi << std::endl;
  NumericVector df(1);
  df[0]  = RcppDeepState_double();
  qs::c_qsave(df,"/home/akhila/fuzzer_packages/fuzzedpackages/bWGR/inst/testfiles/BayesA/inputs/df.qs",
		"high", "zstd", 1, 15, true, 1);
  std::cout << "df values: "<< df << std::endl;
  NumericVector R2(1);
  R2[0]  = RcppDeepState_double();
  qs::c_qsave(R2,"/home/akhila/fuzzer_packages/fuzzedpackages/bWGR/inst/testfiles/BayesA/inputs/R2.qs",
		"high", "zstd", 1, 15, true, 1);
  std::cout << "R2 values: "<< R2 << std::endl;
  std::cout << "input ends" << std::endl;
  try{
    BayesA(y,X,it[0],bi[0],df[0],R2[0]);
  }
  catch(Rcpp::exception& e){
    std::cout<<"Exception Handled"<<std::endl;
  }
}
